vscode/src/chatAgent.ts

```diff
@@ -0,0 +1,214 @@
+import * as vscode from "vscode";
+
+import { Command } from "./common";
+import { Workspace } from "./workspace";
+
+const CHAT_AGENT_ID = "rubyLsp.chatAgent";
+const DESIGN_PROMPT = `
+  You are a domain driven design and Ruby on Rails expert.
+  The user will provide you with details about their Rails application.
+  The user will ask you to help model a single specific concept.
+  Your job is to suggest a model name and attributes to model that concept.
+  Include all Rails \`generate\` commands in a single Markdown shell code block at the end.
+  The \`generate\` commands should ONLY include the type of generator and arguments, not the \`rails generate\` part
+  (e.g.: \`model User name:string\` but not \`rails generate model User name:string\`).
+  NEVER include commands to migrate the database as part of the code block.
+`.trim();
+
+export class ChatAgent implements vscode.Disposable {
+  private readonly agent: vscode.ChatParticipant;
+  private readonly showWorkspacePick: () => Promise<Workspace | undefined>;
+
+  constructor(
+    context: vscode.ExtensionContext,
+    showWorkspacePick: () => Promise<Workspace | undefined>,
+  ) {
+    this.agent = vscode.chat.createChatParticipant(
+      CHAT_AGENT_ID,
+      this.handler.bind(this),
+    );
+    this.agent.iconPath = vscode.Uri.joinPath(context.extensionUri, "icon.png");
+    this.showWorkspacePick = showWorkspacePick;
+  }
+
+  dispose() {
+    this.agent.dispose();
+  }
+
+  // Handle a new chat message or command
+  private async handler(
+    request: vscode.ChatRequest,
+    context: vscode.ChatContext,
+    stream: vscode.ChatResponseStream,
+    token: vscode.CancellationToken,
+  ) {
+    if (this.withinConversation("design", request, context)) {
+      return this.runDesignCommand(request, context, stream, token);
+    }
+
+    stream.markdown(
```

Comment:
If no command is used for the agent, there's really not much we can provide that regular Copilot chat won't.

Here, we only provide specialized behaviour if the user has already initiated a conversation with a previous command.